lldbinit.py

```diff
@@ -9,16 +9,133 @@
+            if frame != original_frame:
+                result = result.Persist()
+            return result
+        frame = frame.get_parent_frame()
+    return None
+
+def sorbet_obj_toString(sorbetObj, error = None):
+    if sorbetObj.TypeIsPointerType():
+        sorbetObj = sorbetObj.Dereference()
+    sorbetType = sorbetObj.GetType()
+    frame = sorbetObj.GetFrame()
+    exprMethods = [sorbetType.GetMemberFunctionAtIndex(i) for i in range(sorbetType.GetNumberOfMemberFunctions())]
+    toStringMethod = next((m for m in exprMethods if m.GetName() == "toString"), None)
+    if toStringMethod is None:
+        if not(error is None):
+            error.SetErrorString("Object of type %s doesn't have a toString method" % sorbetType)
+        return None
+    # Try to retrieve GS and CFG from the current context
+    gs = get_variable_in_frames(frame, ["sorbet::core::GlobalState", "sorbet::core::Context"])
+    if gs is None:
+        if not(error is None):
+            error.SetErrorString("Couldn't find a GS instance in the current frame")
+        return None
+    cfg = get_variable_in_frames(frame, "sorbet::cfg::CFG")
+    if toStringMethod.GetNumberOfArguments() == 2 and cfg is None:
+        if not(error is None):
+            error.SetErrorString("Couldn't find a CFG instance in the current frame")
+        return None
+    globalStateAccess = "%s%s" % (gs.get_expr_path(), ".state" if "sorbet::core::Context" in gs.GetTypeName() else "")
+    command = sorbetObj.get_expr_path()
+    cmd = "(%s).toString(%s)" % (command, globalStateAccess)
+    if toStringMethod.GetNumberOfArguments() == 2:
+        cmd = "(%s).toString(%s, %s)" % (command, globalStateAccess, cfg.get_expr_path())
+    finalExpr = frame.EvaluateExpression(cmd)
+    if not(finalExpr.IsValid()):
+        if not(error is None):
+            error.SetErrorString("No result was returned from expression evaluation")
+        return None
+    return finalExpr
+
+def cmd_dump_sorbet(debugger, command, exe_ctx, result, internal_dict):
+    error = lldb.SBError()
+    error.SetErrorToGenericError()
+    frame = exe_ctx.GetFrame()
+    # Value that we are supposed to dump and retrieve its type to find a toString method
+    expr = frame.EvaluateExpression(command)
+    finalExpr = sorbet_obj_toString(expr, error)
+    if not(finalExpr is None):
+        result.Print("R: %s\n" % finalExpr)
+    else:
+        result.Print("no result\n")
+        result.SetError(error)
+
+def format_sorbet_core_NameRef(valobj, internal_dict, options):
+    id = valobj.GetChildMemberWithName("_id").GetValueAsSigned(-1)
+    if id <= 0:
+        return "Default ID"
+    elif id < 452:
+        # Keep bound check in sync with NAME_LAST_WELL_KNOWN_NAME from autogenerated file Names_gen.h
+        name = valobj.EvaluateExpression("::sorbet::core::Names::DEBUG_NAMES_LOOKUP[%d]" % id)
```

Comment:
Yep I definitely need to harden that a bit

Reply:
Now that it will look up the global variable for the last known ID, we can predicate the whole branch on that being available and thus avoid the issue